src/AppInstallerRepositoryCore/Microsoft/CheckpointIndex.cpp

```diff
@@ -0,0 +1,212 @@
+
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(result.m_dbconn, "CheckpointIndex_createnew");
+
+        // Use calculated version, as incoming version could be 'latest'
+        result.m_version.SetSchemaVersion(result.m_dbconn);
+
+        result.m_interface->CreateTables(result.m_dbconn);
+
+        result.SetLastWriteTime();
+
+        savepoint.Commit();
+
+        return result;
+    }
+
+    std::shared_ptr<CheckpointIndex> CheckpointIndex::OpenOrCreateDefault(GUID guid, OpenDisposition openDisposition)
+    {
+        const auto& indexPath = GetCheckpointIndexPath(guid);
+        AICLI_LOG(Repo, Info, << "Opening checkpoint index");
+
+        try
+        {
+            if (std::filesystem::exists(indexPath))
+            {
+                if (std::filesystem::is_regular_file(indexPath))
+                {
+                    try
+                    {
+                        AICLI_LOG(Repo, Info, << "Opening existing checkpoint index");
+                        return std::make_shared<CheckpointIndex>(CheckpointIndex::Open(indexPath.u8string(), openDisposition));
+                    }
+                    CATCH_LOG();
+                }
+
+                AICLI_LOG(Repo, Info, << "Attempting to delete bad checkpoint index file");
+                std::filesystem::remove_all(indexPath);
+            }
+
+            return std::make_shared<CheckpointIndex>(CheckpointIndex::CreateNew(indexPath.u8string()));
+        }
+        CATCH_LOG();
+
+        return {};
+    }
+
+#ifndef AICLI_DISABLE_TEST_HOOKS
+    std::optional<std::filesystem::path> s_CheckpointIndexDirectoryOverride{};
+    void TestHook_SetCheckpointIndexDirectory_Override(std::optional<std::filesystem::path>&& checkpointIndexDirectory)
+    {
+        s_CheckpointIndexDirectoryOverride = std::move(checkpointIndexDirectory);
+    }
+#endif
+
+    std::filesystem::path CheckpointIndex::GetCheckpointIndexPath(GUID guid)
+    {
+        wchar_t checkpointGuid[256];
+        THROW_HR_IF(E_UNEXPECTED, !StringFromGUID2(guid, checkpointGuid, ARRAYSIZE(checkpointGuid)));
+
+        const auto DefaultIndexDirectoryPath = Runtime::GetPathTo(Runtime::PathName::LocalState) / s_Checkpoints;
+#ifndef AICLI_DISABLE_TEST_HOOKS
+        const auto checkpointIndexDirectory = s_CheckpointIndexDirectoryOverride.has_value() ? s_CheckpointIndexDirectoryOverride.value() : DefaultIndexDirectoryPath;
+#else
+        const auto checkpointIndexDirectory = DefaultIndexDirectoryPath;
+#endif
+
+        if (!std::filesystem::exists(checkpointIndexDirectory))
+        {
+            std::filesystem::create_directories(checkpointIndexDirectory);
+            AICLI_LOG(Repo, Info, << "Creating checkpoint index directory: " << checkpointIndexDirectory);
+        }
+        else
+        {
+            THROW_HR_IF(ERROR_CANNOT_MAKE, !std::filesystem::is_directory(checkpointIndexDirectory));
+        }
+
+        auto indexPath = checkpointIndexDirectory / checkpointGuid;
```

Comment:
The guid should be a directory name and the checkpoint file inside of it can be a fixed name.  There are other files that can be alongside the SQLite database that won't be cleaned up by simply removing the file.